source/vecs/resource.d

```diff
@@ -0,0 +1,101 @@
+module vecs.resource;
+
+version(vecs_unittest)
+{
+	import aurorafw.unit.assertion;
+	import vecs.entity;
+}
+
+
+///
+@safe nothrow @nogc
+private static size_t nextId()
+{
+	static size_t value;
```

Comment:
Are you sure you want to use a static variable here? This makes your code impossible to use in a multithreaded case. `static` variables are stored in thread-local storage.

```d
private static size_t next()
{
    static size_t value;
    
    return value++;
}

void main()
{
    foreach (_; parallel(iota(100)))
    	next().writeln;
}
```

The code above can have repeated numbers, depending on how many threads you have available. And I assume you want unique numbers here, even on a multithreaded case. If you want to have a shared variable across all threads, you may use `__gshared` variables or just mark the variable as `static shared` and use atomic operations, like:

```d
private static size_t next()
{
    shared static size_t value;
    
    import core.atomic : atomicOp;
    return value.atomicOp!"+="(1);
}
```

This will guarantee you uniqueness across all threads with safe locking mechanisms implemented by the runtime.

If you want something rawer you can opt to use `__gshared` but you need to implement your own locking mechanisms. `__gshared` may be useful if you want to have optimizations on the locking process, like, locking it only on certain circumstances or using an extensive locking mechanism.

Reply:
done, also fixed ComponentId.